### ISSUE TITLE
some questions about cross compiling on Linux (without Docker) 

### ISSUE THREAD

**Issue by Clarence-wcz** [microsoft/onnxruntime#1567]
**Describe the bug**

Does cross-compilation only generate static libraries? I generate dynamic libraries when I execute the "./build.sh "command on my PC. But when compiled directly with cmake, it appears that only static libraries are generated.

**Comment by RyanUnderhill** [comment]
We build the static libraries by default, try adding the '--build_shared-lib' parameter to generate dynamic libraries:

    ./build.sh --build_shared_lib

Does that work?

**Comment by Clarence-wcz** [comment]
I compiled according to the official documents. If I followed the method you provided, I could only recognize the local compiler to compile, and could not use arm-gcc. The generated dynamic library is also x86-64 architecture and does not generate arm architecture files. But if I compile directly with cmake, I can compile with arm-gcc, but only generate static libraries.
https://github.com/microsoft/onnxruntime/blob/master/BUILD.md

**Comment by Clarence-wcz** [comment]
> We build the static libraries by default, try adding the '--build_shared-lib' parameter to generate dynamic libraries:
> 
> ```
> ./build.sh --build_shared_lib
> ```
> 
> Does that work?

I compiled according to the official documents. If I followed the method you provided, I could only recognize the local compiler to compile, and could not use arm-gcc. The generated dynamic library is also x86-64 architecture and does not generate arm architecture files. But if I compile directly with cmake, I can compile with arm-gcc, but only generate static libraries.
https://github.com/microsoft/onnxruntime/blob/master/BUILD.md
